src/main/java/com/demonwav/mcdev/platform/canary/CanaryTemplate.java

```diff
@@ -0,0 +1,82 @@
+package com.demonwav.mcdev.platform.canary;
+
+import com.demonwav.mcdev.buildsystem.BuildSystem;
+import com.demonwav.mcdev.buildsystem.gradle.GradleBuildSystem;
+import com.demonwav.mcdev.buildsystem.maven.MavenBuildSystem;
+import com.demonwav.mcdev.platform.AbstractTemplate;
+import com.demonwav.mcdev.util.MinecraftFileTemplateGroupFactory;
+import com.intellij.ide.fileTemplates.FileTemplate;
+import com.intellij.ide.fileTemplates.FileTemplateManager;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.util.Properties;
+
+public class CanaryTemplate extends AbstractTemplate {
+
+    public static void applyMainClassTemplate(@NotNull Project project,
+            @NotNull VirtualFile file,
+            @NotNull String packageName,
+            @NotNull String className) {
```

Comment:
These should be in line with the top `@NotNull`, like the examples in `AbstractTemplate`.